core/src/main/java/com/dylibso/mcpx4j/core/Mcpx.java

```diff
@@ -56,7 +63,35 @@ public Mcpx build() {
             if (httpClientAdapter == null) {
                 httpClientAdapter = new JdkHttpClientAdapter();
             }
-            return new Mcpx(apiKey, baseUrl, jsonDecoder, httpClientAdapter, config);
+            return new Mcpx(apiKey, baseUrl, profileSlug, jsonDecoder, httpClientAdapter, config);
+        }
+
+        /**
+         * @param profile is a list of components for a slug. It may be empty (then it defaults to `~/default`),
+         *                it may contain one single value `"default"`.
+         *                Otherwise, it must contain two components, the first being a username
+         *                and the second being a profile name.
+         *                Neither the profile name nor the username may contain a slash.
+         *                It is also allowed to pass the values `"~", "default"`, yielding `~/default`.
+         */
+        static String profileIdToSlug(String... profile) {
+            if (profile.length == 0) {
+                return "~/default";
+            }
+            if (profile.length == 1) {
+                if (profile[0].equals("default")) {
+                    return "~/default";
+                } else {
+                    throw new IllegalArgumentException("Invalid profile path: " + Arrays.toString(profile));
+                }
```

Comment:
Ooh, we can accept a single-component name and assume it's under the `~/` scope.

```suggestion
                return "~/" + profile[0];
```

(This is the "banana" case in "So an activeProfile of banana will produce the installation url GET /profiles/~/banana/installation" :banana:)

Reply:
will add a test